cmd/bzzup/main.go

```diff
@@ -106,7 +106,7 @@ func (c *client) uploadFile(file string, fi os.FileInfo) (manifest, error) {
 
 func (c *client) uploadDirectory(dir string) (manifest, error) {
 	dirm := manifest{}
-	prefix := filepath.ToSlash(dir) + "/"
+	prefix := filepath.Clean(filepath.ToSlash(dir)) + "/"
```

Comment:
filepath.Clean uses filepath.Separator. This expression will only work on systems where filepath.Separator is `/` because ToSlash converts all occurrences of the separator to `/`. It's better to do

```text
filepath.ToSlash(filepath.Clean(dir))
```